moirae/estimators/offline/scipy.py

```diff
@@ -0,0 +1,41 @@
+"""Estimate the state of health using SciPy optimizers"""
+from scipy.optimize import minimize, OptimizeResult
+
+from moirae.estimators.offline.base import OfflineEstimator
+from moirae.estimators.offline.objectives import Objective
+from moirae.models.base import GeneralContainer, HealthVariable
+
+
+class ScipyMinimizer(OfflineEstimator):
+    """Estimate using SciPy's
+    `minimize <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html>`_ function.
+
+    Args:
+        objective: Objective function to be optimized
+        kwargs: Passed to the minimize function
+    """
+
+    def __init__(self,
+                 objective: Objective,
+                 **kwargs):
+        self.objective = objective
+        self.scipy_kwargs = kwargs
```

Comment:
Good point. I'll provide an code example which shows changing a few of them and emphasize that the page linked in the class docstring explains the keyword arguments. 

The TypedDict is a cool idea, but I'm leery about keeping it up-to-date with SciPy

Reply:
I had not considered that aspect... at the same time, if we don't want to deal with that, can we not just add the current version of scipy in the `.toml` as a dependency?

Reply:
(I say that because, under the same principle, we don't know that `scipy` won't change its structure such that the `minimize` function is actually under a different submodule, for example, even though I know that is extremely unlikely to ever happen haha)

Reply:
I'm more against pinning the scipy version than writing the TypedDict, but writing the typeddict just feels like a lot of tedious work which (there are tons of optimizer options)

Reply:
yes, absolutely, `scipy` is very well documented and I am just being a bit of a cry-baby haha this is perfectly fine 😄